Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -4937,6 +4937,19 @@ void BytecodeGenerator::emitRequireObjectCoercible(RegisterID* value, ASCIILiter
     emitLabel(target.get());
 }
 
+void BytecodeGenerator::emitRequireObjectCoercibleForDestructuring(RegisterID* value, const Identifier* propertyName)
+{
+    Ref<Label> target = newLabel();
+    OpJnundefinedOrNull::emit(this, value, target->bind(this));
+
+    if (propertyName && !propertyName->isNull())
```

Comment:
Does this need to check if it has a public name and/or it's not a symbol?

Reply:
it doesn't need check explicitly, I added tests for it ( https://github.com/WebKit/WebKit/pull/49762/files#diff-c7136627367e6232c602ca6a8fb12a1527c4982345c7f9fba40692d093a76dd3 )